cdap-docs/developers-manual/source/advanced/adapters.rst

```diff
@@ -0,0 +1,140 @@
+  | month                   | int                  |                      |
+  | day                     | int                  |                      |
+  | hour                    | int                  |                      |
+  | minute                  | int                  |                      |
+  |                         |                      |                      |
+  | # Partition Information |                      |                      |
+  | # col_name              | data_type            | comment              |
+  |                         |                      |                      |
+  | year                    | int                  |                      |
+  | month                   | int                  |                      |
+  | day                     | int                  |                      |
+  | hour                    | int                  |                      |
+  | minute                  | int                  |                      |
+  +=======================================================================+
+
+
+Adding a Custom Adapter Type
+============================
+Adapter types are CDAP Applications that contain a special manifest file and are placed
+in an adapter plugins directory. If you wish to write a custom Adapter type, make sure
+your manifest file contains the following properties:
+
+  * CDAP-Adapter-Type: name by which your adapter type will be referenced. 
+  * CDAP-Adapter-Program-Type: schedulable program type of your adapter.
+    Currently, only WORKFLOW is supported.
+  * CDAP-Adapter-Properties: default properties to use for your adapter, formatted
+    as a JSON Object of strings to strings. Properties set when creating an Adapter
+    will be applied on top of the default properties.
+    Adapter properties are passed to the Adapter program as runtime arguments.
+  * CDAP-Source-Type: the source type. Currently only STREAM is supported. 
+  * CDAP-Sink-Type: the sink type. Currently only DATASET is supported. 
+  * CDAP-Sink-Properties: default properties to use for sinks created when an
+    Adapter is created, formatted as a JSON Object of strings to strings. Sink properties
+    set when creating an Adapter will be applied on top of the default properties.
+    When an Adapter is created, if the sink given does not already exist it will be
+    created with the sink properties. The dataset.class properties must be present
+    for dataset sinks.
+
+You can look at the cdap-adapters module in the CDAP project as an example.
+Once you have built your application jar, you must place it in the directory specified
+by the ``app.adapter.dir`` configuration setting.
+This setting defaults to ``/opt/cdap/master/plugins`` for distributed CDAP, and ``plugins``
+for CDAP Standalone. Note that the jar must be present on all CDAP masters. 
+
+Adapters are an experimental feature in CDAP. As such, they are subject to change
```

Comment:
Should we mark this entire feature as Beta? (see http://docs.cask.co/cdap/2.7.0-SNAPSHOT/en/developers-manual/building-blocks/spark-programs.html for an example)